lib/widgets/components/entry/login/login_email_text_field.dart

```diff
@@ -1,26 +1,55 @@
 import 'package:coffeecard/base/strings.dart';
 import 'package:coffeecard/base/style/colors.dart';
-import 'package:coffeecard/cubits/login/login_cubit.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_bloc/flutter_bloc.dart';
 
-class LoginEmailTextField extends StatelessWidget {
-  const LoginEmailTextField();
+class LoginEmailTextField extends StatefulWidget {
+  const LoginEmailTextField({
+    required this.hasError,
+    required this.autoFocusDelay,
+    required this.onSubmit,
+    required this.onChange,
+  });
+
+  final bool hasError;
+  final Duration autoFocusDelay;
+  final void Function(BuildContext, String) onSubmit;
+  final void Function() onChange;
+
+  @override
+  State<LoginEmailTextField> createState() => _LoginEmailTextFieldState();
+}
+
+class _LoginEmailTextFieldState extends State<LoginEmailTextField> {
+  final _controller = TextEditingController();
+  final _focusNode = FocusNode();
+
+  @override
+  void initState() {
+    // Focus the text field after animations have completed
+    () async {
+      await Future.delayed(widget.autoFocusDelay);
```

Comment:
Can there arise a case where this will hang forever?

Reply:
No.

In the case of a really long set delay, the user can always tap the text field to focus the text field themselves

Reply:
Okay. I'll approve